### PR TITLE
Calculate

### PR DESCRIPTION
**Pull request by g00fer**
Nick: Bartek8

### CODE REVIEW

File: module1/homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -2,10 +2,24 @@
 
 int fibonacci_iterative(int sequence) {
     // TODO: Your implementation goes here
-    return 0;
+    int fib1 = 0;
+    int fib2 = 1;
+    int retval;
+
+    if(0 == sequence) return 0;
+    if(1 == sequence) return 1;
```
Comment (DamiXioR) [inline module1/homework/fibonacci/fibonacci.hpp:10]: shorter solution if(sequence <= 1) return sequence;

File: module1/homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -2,10 +2,24 @@
...
+        fib2 = retval;
+    }
+    return retval;
 }
 
 int fibonacci_recursive(int sequence) {
     // TODO: Your implementation goes here
-    return 0;
+    if(0 == sequence) return 0;
+    if(1 == sequence) return 1;
```
Comment (DamiXioR) [inline module1/homework/fibonacci/fibonacci.hpp:23]: the same if(sequence <= 1) return sequence;

File: module1/homework/calculate/calculate.hpp [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+    else if ("divide" == command)
+    {
+        if( 0 != second)
+        {
+            return std::to_string( first / second);
+        }
+        else
+        {
+            return "Division by 0";
+        }
```
[the remark this answers is not in the document]
Comment (anraf1001) [reply module1/homework/calculate/calculate.hpp:28]: The most probable / default option is `second != 0` (or `0 != second`) so in my opinion it's more clear. Anyway I agree that g00fer can delete `else`.

[the remark this answers is not in the document]
Comment (g00fer) [reply module1/homework/calculate/calculate.hpp:28]: I think it is better to have explicit reason for 'return' if multiple 'returns' are used in function. Typically it is better to have only single 'return' statement in a function.

File: module1/homework/fibonacci/fibonacci.hpp (hunk 1 above)
Comment (g00fer) [reply module1/homework/fibonacci/fibonacci.hpp:10]: I translated polish wikipedia directly into software: "Pierwszy wyraz jest równy 0, drugi jest równy 1, każdy następny jest sumą dwóch poprzednich."

File: module1/homework/fibonacci/fibonacci.hpp (hunk 2 above)
Comment (g00fer) [reply module1/homework/fibonacci/fibonacci.hpp:23]: Thank you for the review!

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 1429, 1119, 1430], "points": [2, 5, 2, 6] }

**Pull request closed without merging**